### DISCUSSION TITLE
a quick feature request: Sheduling / Automations. Being able to run prompts at specific times or scheduled intervals would be wonderful 😉

### DISCUSSION THREAD

**Discussion by Carpea03** [Beyond-Better/bb#53] [Ideas]
a quick feature request: Sheduling / Automations. Being able to run prompts at specific times or scheduled intervals would be wonderful 😉

**Comment by cngarrison** [comment]
Interesting idea. Could you please provide an example use-case. 

There are various ways of approaching it. Eg, would each run be a new conversation, or continuing existing conversation. What sort of feedback/confirmation would you expect. 

Having a use-case example will help with designing the feature.

**Comment by andrewcdiggle** [comment]
The general premise I was thinking around was for the delegation of repetitive or time-sensitive tasks without requiring me to manually input a prompt. This could be anything from a research or summarisation task, through to project/codebase reports.

More specifically, I could be running a project that has multiple contributors all saving information into a project folder. I want to write a prompt that analyses the information of that project against strategic objectives and then have that run repeatedly at the end of the week. This scheduling could also work for folder maintenance tasks (maintaining naming conventions, structure etc...) there could be many other applications.

From my point of view, having a way to easily identify prompts that are scheduled would be great (allows easy planning), which may lead to the implementation as new conversation that repeats. But some of this may also come down to whether existing conversations have valuable knowledge (not in the project folders) for executing the task.

Feedback via a in-app notification or via email would be great.

**Comment by Carpea03** [comment]
"maintaining naming conventions, structure etc"
- that is a great idea!

**Comment by cngarrison** [comment]
Would a crontab entry work for you? 

Since `bb` has CLI chat, you could create a crontab entry to send a prompt. 

This would create a new BB conversation once a day (at 5am) to summarize the submissions in "contributors" directory and email the "answer" message.

```
5 05 * * *  cd path/to/project; bb chat -p 'Summarize the submissions in `contributors/` directory, and then move all files to the `done/` directory.' | mail -s "Results of summary submisions" recipient@domain.com
```

You could continue and existing conversation by including `-i <conversationId>` in the `bb chat` command. But I'd be wary of that since long conversations can get expensive. 

The `bb chat` command will start the BB Server (API) automatically if it isn't running already.

If you provide further use-cases I can suggest other ways of structuring the `bb chat` command to handle automated processing.
